File: 2108-find-first-palindromic-string-in-the-array/2108-find-first-palindromic-string-in-the-array.ts
function firstPalindrome(words: string[]): string {
   for (let i = 0; i < words.length; i++) {
        const word = words[i];
        let isPalindromic = true;

        for (let j = 0; j < word.length/2; j++) {
            if (word[j] !== word[word.length - j - 1]) {
                isPalindromic = false;
                break;
            }
        }

        if (isPalindromic) return word;
    }

    return '';
}

File: 2108-find-first-palindromic-string-in-the-array/2108-find-first-palindromic-string-in-the-array.py
class Solution:
    def firstPalindrome(self, words: List[str]) -> str:
        for word in words:
            isPalindromic = True
            n = len(word)

            for j in range(n // 2):
                if word[j] != word[n - j - 1]:
                    isPalindromic = False
                    break
  
            if isPalindromic: return word
        
        return ""